iree/test/e2e/cuda_specific/dynamic_add.mlir

```diff
@@ -0,0 +1,10 @@
+// RUN: iree-run-mlir --iree-input-type=mhlo -iree-hal-target-backends=cuda -function-input="2x4xf32=[[1.0,2.0,3.0,4.0],[-1.0,-2.0,-3.0,-4.0]]" -function-input="2x4xf32=[[5.0,6.0,7.0,8.0],[-5.0,-6.0,-7.0,-8.0]]" %s | IreeFileCheck %s
```

Comment:
Only dynamic add right now? What's blocking for having the rest? We have a bunch dynamic tests in e2e/regression. Could use those.

Reply:
I can add them but I’m not sure it adds a lot. Ideally I would like to reuse the same test so that it doesn’t have to be duplicated but it will need some non trivial changes to the build file/test so I’m just adding this in order to have a simple test exercising dynamic shapes for now

Reply:
Adding a `iree_check_single_backend_test_suite` for CUDA in `iree/test/e2e/regression/BUILD` in parallel to others won't work? (Sorry if this is obvious; but I didn't follow how you set up everything in the CI side. :) )

Reply:
The problem is that the dynamic shape tests are not `iree_check_single_backend_test_suite` they are `iree_lit_test_suite` handling all the backends and I don't want to add the tags specific to CUDA to that. I'm thinking of changing everything to `iree_check_single_backend_test_suite` but I would prefer doing this in a separate PR

Reply:
That's fine for me. :)